src/gapp_login/wechat/endpoints.py

```diff
@@ -86,12 +99,49 @@ async def login_wechat(
     return rv
 
 
+@router.post("/users/update/wxa")
+async def update_wechat_account_info(
+    encrypted_data: str = Form(...),
+    iv: str = Form(...),
+    user: User = Security(require_user),
+):
+    identity = await WeChatIdentity.get(user.get_identity_id())
+    wechat_session_key = identity.profile.get("wechat_session_key")
+    session_key = base64.b64decode(wechat_session_key)
+    encrypted_data = base64.b64decode(encrypted_data)
+    iv = base64.b64decode(iv)
+    decryptor = Cipher(
+        algorithms.AES(session_key), modes.CBC(iv), backend=default_backend(),
+    ).decryptor()
+    data = decryptor.update(encrypted_data) + decryptor.finalize()
+    unpadder = PKCS7(algorithms.AES.block_size).unpadder()
+    data = unpadder.update(data) + unpadder.finalize()
+    data = json.loads(data)
+    open_id = data.pop("openId")
+    unionid = data.pop("unionId", None)
+    wm = data.pop("watermark")
+    if wm["appid"] not in config.WECHAT_CLIENTS:
+        raise HTTPException(status.HTTP_400_BAD_REQUEST, "Bad wechat AppID")
+
+    if open_id != identity.sub:
+        raise HTTPException(status.HTTP_400_BAD_REQUEST, "Wechat openId mismatched")
+
+    await identity.update(wechat_user_info=data, wechat_unionid=unionid,).apply()
+    return dict(data=data)
+
+
+@router.get("/users/wechat")
```

Comment:
(future) authlib-gino 0.4 would probably want to include this in the user_info endpoint, but we could use this as for now

Reply:
Ok~